### PR TITLE
Update S2Width Cut

### LINKED ISSUE #81: Tune S2 Width
Suggested by @skazama to remove potential wall leakage events as in [this plot](https://xe1t-wiki.lngs.infn.it/lib/exe/detail.php?id=xenon%3Akazama%3Aanomalous_bg_sr1&media=xenon:kazama:s2_width_vs_s2_er_leakage.png).

@zhut19 mentioned the cut may be already too tight, however looking at Figs. A12 and A13 in @weiyuehuan's [tuning note](https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:xenon1t:yuehuan:analysis:1sciencerun_s2width#appendix_3bug_of_this_cut_in_laxlichenssr1), seems like the previous straight line extrapolation can be tightened.

### LINKED ISSUE #105: Update to S1SingleScatter
@jpienaar13 working on this. Should kill one of our anomalous Rn220 events from SR1.

### PR DESCRIPTION
**Pull request by zhut19**
Update the S2Width Cut, which is formulated according to Chi2 distribution.
[Note](https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:xenon1t:sim:notes:tzhu:width_cut_tuning#toy_fax_simulation)

Fixes #81 

Fixes #105

### CODE REVIEW

**Review by sreichard — APPROVED** [review]

**Review by pdeperio — CHANGES_REQUESTED** [review]
Should this instead be implemented in ```sciencerun0.py``` then inherited in ```sciencerun1.py``` ?

(Also, please fix ```codacy``` issues.)

File: lax/lichens/sciencerun1.py [hunk 1]
```diff
@@ -177,66 +175,19 @@ class S2PatternLikelihood(StringLichen):
...
+    version = 5
+    diffusion_constant = 29.35 * ((units.cm)**2) / units.s
+    v_drift = 1.335 * (units.um) / units.ns
+    scg = 21.3 # s2_secondary_sc_gain in pax config
+    scw = 229.58  # s2_secondary_sc_width median
+    SigmaToR50 = 1.349
+
+class S1SingleScatter(sciencerun0.S1SingleScatter):
+    from scipy.stats import chi2
     version = 3
```
Comment (pdeperio) [inline lax/lichens/sciencerun1.py]: Increment ```S1SingleScatter``` version?

File: lax/lichens/sciencerun1.py [hunk 2]
```diff
@@ -177,66 +175,19 @@ class S2PatternLikelihood(StringLichen):
...
-            return df
-
+class S2Width(sciencerun0.S2Width):
+    from scipy.stats import chi2
+    version = 4
+    diffusion_constant = 29.35 * ((units.cm)**2) / units.s
+    v_drift = 1.335 * (units.um) / units.ns
+    scg = 21.3 # s2_secondary_sc_gain in pax config
+    scw = 229.58  # s2_secondary_sc_width median
+    SigmaToR50 = 1.349
```
Comment (pdeperio) [inline lax/lichens/sciencerun1.py]: This line necessary if you're not changing it?

**Review by pdeperio — APPROVED** [review]

File: lax/lichens/sciencerun1.py (hunk 2 above)
Comment (zhut19) [reply lax/lichens/sciencerun1.py]: No, they are just overwriting the sr0 values here.

**Review by pdeperio — CHANGES_REQUESTED** [review]
Request from @skazama to check against a wall leakage background control sample (provided by @AlexElykov?), and re-optimize accordingly if necessary.

**Comment by pdeperio** [comment]
Fixes #105 too i think

**Comment by zhut19** [comment]
The wall leakage background data provided by @jingqiangye have no specific feature in this cut space. As well as in S1/S2 pattern, S2 single scatter, S2 area fraction top.
wall_leakage_cut_check

**Review by pdeperio — APPROVED** [review]

**Pull request merged**